src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { NewCommentForm } from '../NewCommentForm';
+import { Post } from '../../types/Post';
+import { client } from '../../utils/fetchClient';
+import { Comment } from '../../types/Comment';
+
+type Props = {
+  post: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [areCommentsLoading, setAreCommentsLoading] = useState(false);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const [isCommentFormOpen, setIsCommentFormOpen] = useState(false);
+
+  const loadComments = async (postId: number) => {
+    try {
+      setComments(null);
+      setAreCommentsLoading(true);
+
+      const loadedComments: Comment[] = await client.get(
+        `/comments?postId=${postId}`,
+      );
+
+      setComments(loadedComments);
+    } catch {
+      setHasLoadingError(true);
+    } finally {
+      setAreCommentsLoading(false);
+    }
+  };
+
+  const deleteComment = async (commentId: number) => {
+    try {
+      if (comments) {
+        setComments(comments.filter(comment => comment.id !== commentId));
+      }
```

Comment:
It's better to do this after API request. Because if your request will fail with an error you don't need to remove this comment from the state.

Reply:
I agree it's not a good idea deleting a comment before it was deleted on server. But Readme tells to do so. And tests fail otherwise((

<img width="551" alt="Снимок экрана 2023-05-04 в 18 28 15" src="https://user-images.githubusercontent.com/109100085/236255453-12d869a6-d26d-4a33-808b-ba429e2bc817.png">